### ISSUE TITLE
About 'intersection' function

### ISSUE THREAD

**Issue by jianglinghan** [networkx/networkx#3265]
Why intersection need the nodes lists to be  the same?

In my case, 
```
G1 = nx.Graph()
G1.add_edges_from([(1, 2), (1, 3), (1, 4)])
G2 = nx.Graph()
G2.add_edges_from([(1, 2), (1, 3)]) 
```
I want to get the intersection between G1 and G2, but it can't be done by 'intersection'.

Can anyone explain the motivation of this requirement ?

**Comment by renan-cunha** [comment]
I am wondering this too, for me, the intersection operation should be done especially with graphs with different nodes.

**Comment by dschult** [comment]
That code was put into NetworkX a long time ago. The question is what is the definition of the intersection of two graphs. The NetworkX function should return that.  From what I'm finding with a quick search, the resulting graph should have nodes that are the intersection of the two node sets and edges that are also the intersection of the two edge sets.

**Comment by hagberg** [comment]
Needs pull request to update code for intersection of graphs to allow intersection of graphs with different node sets.

**Comment by rossbar** [comment]
Closed by #4588
